File: template/grace_template.py
NL = chr(10)
DQ = chr(34)

s = "$STR$"

def decl_func(name, body):
  exec(f'def {name}():{NL}  {body}', globals())

decl_func("write_file", "open('Grace_kid.py', 'w').write($ARG$)")
decl_func("main", "write_file()")

# ANYWAY IT WORKS!
if __name__ == '__main__':
  main()

File: template/grace_template.c
#include <err.h>
#include <stdio.h>

#define meta "$STR$"

#define DECL_MAIN() \
  int main(void) {  \
    write_file();   \
    return 0;       \
  }

#define WRITE_FILE(filename)             \
  void write_file(void) {                \
    FILE *handle = fopen(filename, "w"); \
                                         \
    if (handle == NULL) err(1, "fopen"); \
    fprintf(handle, $ARG$);              \
  }

/*
  ANYWAY IT WORKS!
*/
WRITE_FILE("Grace_kid.c")
DECL_MAIN()